server/genserver.go

```diff
@@ -460,6 +481,24 @@ func (h handler) {{.Op}}Handler(ctx context.Context, w http.ResponseWriter, r *h
 		return
 	}
 
+	{{if .HasPaging}}
+	if !swag.IsZero(nextPageID) {
+		{{if .SingleStringPathParameter}}
+		{{.InputVarName}} = nextPageID
+		w.Header().Set("X-Next-Page-Path", models.{{.Op}}InputPath({{.InputVarName}})
```

Comment:
Hmm I think maybe we should just disallow paging with `SingleStringPathParameter` since it's pretty weird to have the page parameter in the path. What do you think?